util/detectOS.ts

```diff
@@ -0,0 +1,24 @@
+import type { UserOS } from '../types/userOS';
+
+export const detectOS = (): UserOS => {
+  return (
+    // Since `navigator.appVersion` is deprecated, we use the `userAgent``
+    // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/appVersion
+    detectOsInUserAgent(navigator.userAgent) || 'UNKNOWN'
+  );
+};
+
+export const detectOsInUserAgent = (userAgent: string): UserOS => {
+  const osMatch = userAgent.match(/(Win|Mac|Linux)/);
+  const os = (osMatch && osMatch[1]) || '';
+  switch (os) {
```

Comment:
No need to assign a default string. If this is undefined, well, all good.

```suggestion
  switch (osMatch && osMatch[1]) {
```